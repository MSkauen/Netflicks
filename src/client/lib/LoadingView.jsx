import React from "react";
import spinner  from "../../shared/img/Spinner.svg"

export function LoadingView() {
  return (
      <>
        <div id="error-view">
            <img src={spinner} alt="Loading" />
        </div>
      </>
  );
}
