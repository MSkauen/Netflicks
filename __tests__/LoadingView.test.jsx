import TestRenderer from 'react-test-renderer'; // ES6
import {LoadingView} from "../src/client/lib/LoadingView";
import React from "react";
import ReactDOM from "react-dom";
import { describe, expect, it } from "@jest/globals";
import { act } from "react-dom/test-utils";

describe("loading view", () => {
    it("show loading view", () => {
        const view = TestRenderer.create(<LoadingView/>);
            expect(view.toJSON()).toMatchSnapshot();
    })
    it("show loading view on dom", () => {
        const container = document.createElement("div");
        document.body.appendChild(container);
        act(() => ReactDOM.render(<LoadingView/>, container))
            expect(container.innerHTML).toMatchSnapshot();
    })
})
