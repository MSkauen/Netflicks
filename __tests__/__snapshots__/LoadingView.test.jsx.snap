// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading view show loading view 1`] = `
<div
  id="error-view"
>
  <img
    alt="Loading"
    src=""
  />
</div>
`;

exports[`loading view show loading view on dom 1`] = `"<div id="error-view"><img src="" alt="Loading"></div>"`;
